### ISSUE TITLE
Rewind not working, causes Kodi to fast-forward instead

### ISSUE THREAD

**Issue by arabelle** [jellyfin/jellyfin-kodi#390]
**Describe the bug**

On our Kodi setup, pressing rewind on the UI causes the video to skip ahead.

**To Reproduce**

1. Open a video on Kodi to play.
2. Press (|<) button
4. See video skip forward

**Expected behavior**

Pressing rewind should make the timestamp go backwards instead of forwards.

**Logs**

**Screenshots**

**System (please complete the following information):**
 - OS: [LibreElec](https://libreelec.tv/) 9.2.5
 - Jellyfin Version: 10.5.3
 - Kodi Version: 18.8.0 (leia)
 - Addon Version: 0.5.9+py2
 - Playback Mode: Native

**Additional context**

Could be because of this line https://github.com/jellyfin/jellyfin-kodi/blob/master/jellyfin_kodi/player.py#L303

**Comment by mcarlton00** [comment]
I can't replicate this in addon mode.  Which means it's either something native mode specific (which I thought was using kodi proper's controls) or there's something very weird not behaving in your system.  Do you have any more details?

**Comment by JeremyBolster** [comment]
I was able to replicate this in add-on mode; if and only if transcoding does not take place. 
Related Issues:
* [Emby](https://emby.media/community/index.php?/topic/69668-emby-for-kodi-timeshift-not-working/&/topic/69668-emby-for-kodi-timeshift-not-working/?hl=timeshift)
* [Emby](https://emby.media/community/index.php?/topic/73740-timeshift-not-working/)
* [Kodi](https://forum.kodi.tv/showthread.php?tid=344193)

### Digging 

Now the fun part, digging way into this. I found the value `input stream reports it is not seekable` in the logs which led me to the code [here](https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp) and an issue from a long time ago [here](https://forum.kodi.tv/showthread.php?tid=189963&page=5). By the looks of this thread, the issue is caused by the Jellyfin server setting headers. But why this issue is only popping up in Leia `18.2+` releases? It doesn't really make any sense. The encoding check code in Kodi has existed since the first port of the XBMC/Kodi see [here](https://github.com/xbmc/xbmc/blob/master/xbmc/filesystem/CurlFile.cpp#L1066).

#### Why now? (Leia)

tl;dr [HTTP2](https://github.com/xbmc/xbmc/commit/8839957437f2c1557c5810aff2af820e50826370)

Looks like support for HTTP2 in kodi was enabled in `18.2-Leia` (See the commit above). This is supported by [Zriel](https://forum.kodi.tv/showthread.php?tid=344193&pid=2864495#pid2864495) who tested on `18.1`, `18.2`, and `18.3` and only noticed the issue on `18.2+`
Does that make it a Kodi issue? No, probably not. `Transfer-Encoding` should not be set by the server when using http2 according to the RFC. However, when I tried setting the advanced setting to [disable http2](https://github.com/xbmc/xbmc/pull/16601/files) there was no change in behavior. Trying to connect to the direct stream with curl on the cmdline while specifying HTTP1.1 showed that curl will upgrade the connection automatically even if you set the initial handshake to a lower version of HTTP. I didn't have time to setup a proxy or build a new docker image for jellyfin to strip http2 out but I suspect that will solve the issue. 

# tl;dr
## Workaround
If using add-on mode, force transcoding.
## The Fix
I think the fix needs to be made in Jellyfin. `Transfer-Encoding` should not be set in the http2 headers. 

# Apendix
##### Advanced Setting
```xml
<advancedsettings>
    <network>
        <disablehttp2>true</disablehttp2>
    </network>
</advancedsettings>
```
##### Release Notes
The release notes of Kodi make no mention of http2 support in Leia. 

##### Native Mode
Oh I also found a weird bug that if you install Jellyfin in add-on mode and switch to native, you may still be accessing the files in add-on mode triggering this bug.

**Comment by mcarlton00** [comment]
I stand corrected, it could also be an issue in the server.  Didn't think of that.  Nice detective work and writeup!  It also explains why I can't replicate it since my test boxes are on my lan and not going through my reverse proxy.

The only place I see `Transfer-Encoding` referenced in the JF server repo is https://github.com/jellyfin/jellyfin/blob/fc827a7b25114e0b987cf913702a242e2f9b89a2/debian/po/templates.pot#L18.  Seems like we'll need somebody from @jellyfin/server to look at that part to verify.

In reference to the switching mode bug: if you didn't resync the database after changing modes that's bound to happen.  The paths to the URL of each file is stored in the database, so if you had synced on Add-On mode it's still going to contain http URLs instead of a direct path to the network share.  Yet another thing I need to slap a big disclaimer on at some point.
